cardano-db-sync/src/Cardano/DbSync/Api/Ledger.hs

```diff
@@ -54,19 +54,19 @@ bootStrapMaybe ::
 bootStrapMaybe syncEnv = do
   bts <- liftIO $ readTVarIO (envBootstrap syncEnv)
   when bts $ do
-    migrateBootstrapUTxO syncEnv emptyTxCache -- TODO: hardcoded to empty
-
-newtype TxCache = TxCache {txIdCache :: Map ByteString DB.TxId}
-
-emptyTxCache :: TxCache
-emptyTxCache = TxCache mempty
+    case envCache syncEnv of
+      NoCache -> migrateBootstrapUTxO syncEnv
+      ActiveCache ci -> do
+        -- TODO: hardcoded to empty
```

Comment:
I think it's a good idea reusing the cache for this separate functionality

Reply:
I'm not sure I understand what you mean, what is "reusing the cache"? The exiting cache or the new cache which is part of `CacheStatus`. It's all honesty it's not directly clear what `TxCache` was used for and why it was just populated as empty because I couldn't see where it was been populated after.

Reply:
I liked the idea of using the `TxCache` for the UTxO bootsrap, which is separate from https://github.com/IntersectMBO/cardano-db-sync/issues/1727. We get two for the cost of one.

Btw the previous definition of `TxCache` was an unimplemented optimization, so you can ignore.

Reply:
ahhhh that's why nothing was happening to it. Wasn't sure if it was voodoo magic or something 😅